channels/rail/client/rail_orders.c

```diff
@@ -650,7 +650,9 @@ UINT rail_order_recv(railPlugin* rail, wStream* s)
 		case RDP_RAIL_ORDER_EXEC_RESULT:
 			{
 				RAIL_EXEC_RESULT_ORDER execResult;
-				return rail_recv_exec_result_order(rail, &execResult, s);
+				error = rail_recv_exec_result_order(rail, &execResult, s);
+				free(execResult.exeOrFile.string);
```

Comment:
This field is not actually initialized in execResult. If `rail_recv_exec_result_order()` fails before filling in the string (e.g., in `rail_read_server_exec_result_order()`) the we'll get heap corruption here.

I'd recommend to zero-initialize execResult (and other structures in this function, while we're at it). I guess you can try adding `= { 0 }` to declarations, but if that fails then explicit `ZeroMemory` calls should do the trick.